astroquery/mast/missions.py

```diff
@@ -0,0 +1,201 @@
+
+        Parameters
+        ----------
+        response : `~requests.Response`
+            `~requests.Response` objects.
+        verbose : bool
+            (presently does nothing - there is no output with verbose set to
+            True or False)
+            Default False.  Setting to True provides more extensive output.
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+
+        print(response)
+        return self._service_api_connection._parse_result(response, verbose, data_key='results')
+
+    @class_or_instance
+    def query_region_async(self, coordinates, radius=3*u.arcmin, **kwargs):
+        """
+        Given a sky position and radius, returns a list matching datasets.
+
+        Parameters
+        ----------
+        coordinates : str or `~astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `~astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.2 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used. Defaults to 0.2 deg.
+        mission : str, optional
+            Default HST.
+            The mission to be queried.
+        **kwargs
+            Other mission-specific keyword args.
+            These can be found in the (service documentation)[https://mast.stsci.edu/api/v0/_services.html]
+            for specific catalogs. For example one can specify the magtype for an HSC search.
+
+        Returns
+        -------
+        response : list of `~requests.Response`
+        """
+
+        # Put coordinates and radius into consistant format
+        coordinates = commons.parse_coordinates(coordinates)
+
+        # if radius is just a number we assume degrees
+        radius = coord.Angle(radius, u.arcmin)
+
+        # basic params
+        params = {'target': [f"{coordinates.ra.deg} {coordinates.dec.deg}"],
+                  'radius': radius.arcmin,
+                  'radius_units': 'arcminutes'}
+
+
+        # adding additional user specified parameters
+        for prop, value in kwargs.items():
+            params[prop] = value
+
+        return self._service_api_connection.service_request_async(self.service, params, use_json=True)
+
+    @class_or_instance
+    def query_criteria_async(self, **criteria):
+        """
+        Given an set of filters, returns a list of catalog entries.
+
+        Parameters
+        ----------
+        **criteria
+            Criteria to apply. At least one non-positional criteria must be supplied.
+            Valid criteria are coordinates, objectname, radius (as in `query_region` and `query_object`),
+            and all fields listed in the column documentation for the mission being queried.
```

Comment:
Can you point to the specific documentation? Or at least where they can find that out?

Reply:
@bsipocz So, this interface will support multiple missions. Right now we support one (hst) but two (classy, ullyses) are being worked on and would be released soon. Each mission has it's own set of fields that can be used in searching mission data. We don't want to change/update docstrings every time a new mission is released/ fields added to existing mission.

Reply:
@syed-gilani  suggest pointing to the get_column function here, since that will give the correct results for the mission the user has set.

Reply:
@ceb8 I have added a method get_column_list() that returns all the field names that users can pass in **kwargs for a given mission.